model/src/test/kotlin/licenses/LicenseInfoResolverTest.kt

```diff
@@ -442,22 +442,59 @@ class LicenseInfoResolverTest : WordSpec() {
                     createLicenseInfo(
                         id = pkgId,
                         declaredLicenses = setOf("$apacheLicense or $gplLicense", mitLicense),
+                        detectedLicenses = listOf(
+                            Findings(
+                                provenance = provenance,
+                                licenses = mapOf(
+                                    "$gplLicense OR $bsdLicense" to listOf(
+                                        TextLocation("LICENSE", 1),
+                                        TextLocation("LICENSE", 21)
+                                    ),
+                                    bsdLicense to listOf(
+                                        TextLocation("LICENSE", 31),
+                                        TextLocation("LICENSE", 41)
+                                    )
+                                ).toFindingsSet(),
+                                copyrights = setOf(
+                                    CopyrightFinding(
+                                        "Copyright GPL 2.0 OR BSD Zero Clause",
+                                        TextLocation("LICENSE", 1)
+                                    ),
+                                    CopyrightFinding("Copyright BSD Zero Clause", TextLocation("LICENSE", 31))
+                                ),
+                                licenseFindingCurations = emptyList(),
+                                pathExcludes = emptyList(),
+                                relativeFindingsPath = ""
+                            )
+                        )
                     )
                 )
                 val resolver = createResolver(licenseInfos)
 
+                val detectedSpdxExpressions =
+                    licenseInfos[0].detectedLicenseInfo.findings.flatMap { it.licenses }.map { it.license }
+                        .toTypedArray()
+
                 val result: ResolvedLicenseInfo = resolver.resolveLicenseInfo(pkgId)
                 result.id shouldBe pkgId
-                result should containOnlyLicenseSources(LicenseSource.DECLARED)
-                result should containLicenseExpressionsExactly(licenseInfos[0].declaredLicenseInfo.processed.spdxExpression)
-                result should containLicensesExactly(apacheLicense, gplLicense, mitLicense)
-                result should containNoLicenseLocations()
-                result should containNoCopyrights()
+                result should containOnlyLicenseSources(LicenseSource.DECLARED, LicenseSource.DETECTED)
+                result should containOriginalLicenseExpressionsBySourceExactly(
+                    setOf(LicenseSource.DECLARED),
+                    licenseInfos[0].declaredLicenseInfo.processed.spdxExpression
+                )
+                result should containOriginalLicenseExpressionsBySourceExactly(
+                    setOf(LicenseSource.DETECTED),
+                    *detectedSpdxExpressions
```

Comment:
As in the previous commit, for the readability of the test and to reduce dependencies on other code it would be better to list the expressions here and in the check above explicitly.